src/main/java/org/osc/manager/ism/api/IsmDeviceApi.java

```diff
@@ -138,80 +189,217 @@ public Void call() throws Exception {
     @Override
     public String createDeviceMember(final String name, String ipAddress, String vserverIpAddress, String contactIpAddress,
             String gateway, String prefixLength) throws Exception {
-        return txControl.required(new Callable<DeviceMember>() {
+
+        DeviceEntity device = new DeviceEntity();
+        DeviceMemberEntity deviceMember = new DeviceMemberEntity();
+        deviceMember.setParent(device);
+        device.setId(null);
+        deviceMember.setName(name);
+        return createDeviceMember(deviceMember, null, this.vs.getId());
+    }
+
+    public String createDeviceMember(DeviceMemberEntity entity, Long parentId, Long vsId) throws Exception {
+
+        return this.txControl.required(new Callable<DeviceMemberEntity>() {
+
             @Override
-            public DeviceMember call() throws Exception {
-                VSSDevice device = em.find(VSSDevice.class, vs.getId());
-                
-                DeviceMember dm = new DeviceMember();
-                dm.setParent(device);
-                dm.setName(name);
-                em.persist(dm);
-                return dm;
+            public DeviceMemberEntity call() throws Exception {
+
+                CriteriaBuilder criteriaBuilder = IsmDeviceApi.this.em.getCriteriaBuilder();
+                CriteriaQuery<DeviceEntity> query = criteriaBuilder.createQuery(DeviceEntity.class);
+                Root<DeviceEntity> r = query.from(DeviceEntity.class);
+                if (parentId != null) {
+                    query.select(r)
+                    .where(criteriaBuilder.and(criteriaBuilder.equal(r.get("id"), parentId)));
+                } else {
+                    query.select(r).where(
+                            criteriaBuilder.and(criteriaBuilder.equal(r.get("vsId"), vsId)));
+                }
+
+                List<DeviceEntity> devices = IsmDeviceApi.this.em.createQuery(query).getResultList();
+                if (devices.isEmpty()) {
+
```

Comment:
extra ln. Please take an overall look at the code and check for consistent indentation, spaces and line break.s